src/v2/style-guide/index.md

```diff
@@ -0,0 +1,1805 @@
+      // ...
+    }
+  }
+}
+```
+
+``` js
+var myGreatMixin = {
+  // ...
+  methods: {
+    _update: function () {
+      // ...
+    }
+  }
+}
+```
+
+``` js
+var myGreatMixin = {
+  // ...
+  methods: {
+    $update: function () {
+      // ...
+    }
+  }
+}
+```
+
+``` js
+var myGreatMixin = {
+  // ...
+  methods: {
+    $_update: function () {
+      // ...
+    }
+  }
+}
+```
+
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+``` js
+var myGreatMixin = {
+  // ...
+  methods: {
+    $_myGreatMixin_update: function () {
+      // ...
+    }
+  }
+}
+```
+{% raw %}</div>{% endraw %}
+
+
+
+## Priority B Rules: Strongly Recommended (Improving Readability)
+
+
+
+### Component files <sup data-p="b">strongly recommended</sup>
+
+**Whenever a build system is available to concatenate files, each component should be in its own file.**
+
+This helps you to more quickly find a component when you need to edit it or review how to use it.
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+``` js
+Vue.component('TodoList', {
+  // ...
+})
+
+Vue.component('TodoItem', {
+  // ...
+})
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+```
+components/
+|- TodoList.js
+|- TodoItem.js
+```
+
+```
+components/
+|- TodoList.vue
+|- TodoItem.vue
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Single-file component filename casing <sup data-p="b">strongly recommended</sup>
+
+**Filenames of [single-file components](single-file-components.html) should either be always PascalCase or always kebab-case.**
+
+PascalCase works best with autocompletion in code editors, as it's consistent with how we reference components in JS(X) and templates, wherever possible. However, mixed case filenames can sometimes create issues on case-insensitive filesystems, which is why kebab-case is also perfectly acceptable.
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+```
+components/
+|- mycomponent.vue
+```
+
+```
+components/
+|- myComponent.vue
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+```
+components/
+|- MyComponent.vue
+```
+
+```
+components/
+|- my-component.vue
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Base component names <sup data-p="b">strongly recommended</sup>
+
+**Base components (a.k.a. presentational, dumb, or pure components) that apply app-specific styling and conventions should all begin with a specific prefix, such as `Base`, `App`, or `V`.**
+
+{% raw %}
+<details>
+<summary>
+  <h4>Detailed Explanation</h4>
+</summary>
+{% endraw %}
+
+These components lay the foundation for consistent styling and behavior in your application. They can only contain:
+
+- HTML elements,
+- other `Base`-prefixed components, and
+- 3rd-party UI components.
+
+They'll never contain:
+
+- global state (e.g. from a Vuex store)
+-
```

Comment:
What's this hyphen for?

Reply:
For adding more items that I never needed to add. Removing this - thanks!